test/functional/page_objects/common_page.ts

```diff
@@ -64,96 +70,128 @@ export function CommonPageProvider({ getService, getPageObjects }) {
       }
     }
 
-    getHostPort() {
+    /**
+     * Returns Kibana host URL
+     */
+    public getHostPort() {
       return getUrl.baseUrl(config.get('servers.kibana'));
     }
 
-    getEsHostPort() {
+    /**
+     * Returns ES host URL
+     */
+    public getEsHostPort() {
       return getUrl.baseUrl(config.get('servers.elasticsearch'));
     }
 
     /**
-     * @param {string} appName As defined in the apps config
-     * @param {string} subUrl The route after the hash (#)
+     * Logins to Kibana as default user and navigates to provided app
+     * @param appUrl Kibana URL
      */
-    async navigateToUrl(appName, subUrl, {
-      basePath = '',
-      ensureCurrentUrl = true,
-      shouldLoginIfPrompted = true,
-      shouldAcceptAlert = true
-    } = {}) {
-      // we onlt use the pathname from the appConfig and use the subUrl as the hash
-      const appConfig = {
-        pathname: `${basePath}${config.get(['apps', appName]).pathname}`,
-        hash: `/${appName}/${subUrl}`,
-      };
-
-      const appUrl = getUrl.noAuth(config.get('servers.kibana'), appConfig);
-
-      await retry.try(async () => {
-        await CommonPage.navigateToUrlAndHandleAlert(appUrl, shouldAcceptAlert);
-        const currentUrl = shouldLoginIfPrompted ? await this.loginIfPrompted(appUrl) : await browser.getCurrentUrl();
+    private async loginIfPrompted(appUrl: string) {
+      let currentUrl = await browser.getCurrentUrl();
+      log.debug(`currentUrl = ${currentUrl}\n    appUrl = ${appUrl}`);
+      await find.byCssSelector('[data-test-subj="kibanaChrome"]', 6 * defaultFindTimeout); // 60 sec waiting
```

Comment:
I decreased timeout from **4** min to **1** min because it makes sense to fail and retry from scratch with url navigation rather than wait so long for the page to be loaded

Reply:
Curious, why private?

Reply:
Well, it is used only in the same class as a helper function. For the tests we expose use navigation functions, e.g. `navigateToApp` and use `{ shouldLoginIfPrompted: boolean }` to manage login. I'm open to change it, but not sure we want it to be used directly